chore: Add example of usage of low_wasm_memory_hook (#1112)
dfinity/examples: rust/low_wasm_memory/README.md

# Low Wasm memory hook

> Internet Computer can automatically execute special type of function called low Wasm memory hook, which runs when the available Wasm memory of the canister falls below 'wasm-memory-threshold'.

Reviewer: ```suggestion
The Internet Computer can automatically execute a special type of function called low Wasm memory hook, which runs when the available Wasm memory of the canister falls below the 'wasm-memory-threshold'.
```

> Unlike other blockchains, the Internet Computer can automatically execute special type of function called low Wasm memory hook, which runs when the available Wasm memory of the canister falls below 'wasm-memory-threshold'.

Reviewer 2: > Unlike other blockchains

Is this part really true? Should we just avoid saying it?

This example demonstrates the ways of using low Wasm memory hook on the Internet Computer.

> The example consists of canister named `low_wasm_memory_hook` implementing the functionality that it increases usage of Wasm memory in every 'heartbeat' execution, until the low Wasm memory hook is runned.

Reviewer: ```suggestion
The example consists of a canister named `low_wasm_memory_hook` implementing the functionality that it increases usage of Wasm memory in every 'heartbeat' execution, until the low Wasm memory hook is run.
```

> The example consists of canister named `low_wasm_memory_hook` both implementing the functionality that it increases usage of Wasm memory in every 'heartbeat' execution, until the low Wasm memory hook is runned.

Reviewer 2: ```suggestion
The example consists of canister named `low_wasm_memory_hook` implementing the functionality that it increases usage of Wasm memory in every 'heartbeat' execution, until the low Wasm memory hook is runned.
```

      heartbeat: http://127.0.0.1/...
      timer: http://127.0.0.1/...
```

- #### Setp 4: Update canister settings

Update canister settings:

```sh
dfx canister update-settings low_wasm_memory_hook --wasm-memory-limit 3000000 --wasm-memory-threshold 2000000
```

and test that setting are correctly updated:

```sh
dfx canister status low_wasm_memory_hook
```

Example output:

```sh
% dfx canister status low_wasm_memory_hook
Canister status call result for low_wasm_memory_hook.
Status: Running
Controllers: bnz7o-iuaaa-aaaaa-qaaaa-cai k7ujo-pl7jf-zqnnx-gdutf-uk5ck-4ngld-xq5hi-276ph-32z4y-ckaue-uae
Memory allocation: 0 Bytes
Compute allocation: 0 %
Freezing threshold: 2_592_000 Seconds
Idle cycles burned per day: 1_508_309 Cycles
Memory Size: 1_918_740 Bytes
Balance: 3_061_265_520_627 Cycles
Reserved: 0 Cycles
Reserved cycles limit: 5_000_000_000_000 Cycles
Wasm memory limit: 3_000_000 Bytes
Wasm memory threshold: 2_000_000 Bytes
Module hash: 0x715518791ccdc6b26cdc66e1c8405f2965208caf611ef4b698c612419dcbbf75
Number of queries: 0
Instructions spent in queries: 0
Total query request payload size: 0 Bytes
Total query response payload size: 0 Bytes
Log visibility: controllers
```

> In example above we set 'wasm-memory-limit' on 3MB and 'wasm-memory-threshold' on 2MB. Hence whenever Wasm memory

Reviewer: ```suggestion
In the example above we set the 'wasm-memory-limit' to 3MB and 'wasm-memory-threshold' to 2MB. Hence whenever the Wasm memory
```

> used by the canister is above 1MB (in other words reamining Wasm memory is less than 'wasm-memory-threshold') 
> low Wasm memory hook will run.

Reviewer: ```suggestion
used by the canister is above 1MB (in other words, the remaining Wasm memory is less than 'wasm-memory-threshold') 
the low Wasm memory hook will run.
```
